src/core/vector/qgsvectorlayerfeatureiterator.cpp

```diff
@@ -1045,11 +1049,13 @@ void QgsVectorLayerFeatureIterator::FetchJoinInfo::addJoinedAttributesCached( Qg
 
 void QgsVectorLayerFeatureIterator::FetchJoinInfo::addJoinedAttributesDirect( QgsFeature &f, const QVariant &joinValue ) const
 {
+#if 0 // this is not thread safe -- we cannot access the layer here as this will be called from non-main threads.
```

Comment:
Why keep it?

Reply:
it'd be a nice optimisation to revive at some stage, if the required api is added